### DISCUSSION TITLE
flink db2 cdc default value error

### DISCUSSION THREAD

**Discussion by 1942309883** [apache/flink-cdc#3410] [Q&A]
I am using flink db2-cdc to sync database in real time,but fails to handle default values in schema when is making the snapshot.
![image](https://github.com/apache/flink-cdc/assets/18555755/2959745b-0952-4a27-a741-c03d13c47061)
After digging deeper into the problem, I found that this seems to be a bug in debezium and was fixed in 2.0.0.CR1(https://issues.redhat.com/browse/DBZ-4990).
The latest flink3.1 uses debezium version 1.9.8.Final. Is there any way to solve this problem in this version, or can we only wait for flink cdc to support debezium2?

**Comment by yuxiqian** [comment]
Hi Luke,
There's a plan to upgrade Debezium base version to 2.5.4.Final after Flink CDC drops Java 8 support since Debezium 2.x forcefully requires JDK 11, but this won't happen very quickly.

For now I think we can backport this patch to 1.9.8.Final and see if it could solve your problem. Could you please open a [JIRA](https://issues.apache.org/jira/projects/FLINK/summary) ticket to trace this issue?

**Reply by 1942309883** [reply]
OK, I am applying for a JIRA account, I will submit this question if the application is approved, thank you for your reply.

**Reply by 1942309883** [reply]
Issue added to: (https://issues.apache.org/jira/browse/FLINK-35572)

**Reply by yuxiqian** [reply]
Thanks! Will investigate this.

**Reply by 1942309883** [reply]
By interpreting the debezium source code, I found that Debezium-connector-db2-2.0 fixed this bug by implementing the io.debezium.connector.db2.Db2DefaultValueConverter class.
![image](https://github.com/apache/flink-cdc/assets/18555755/5aa09645-40dd-4dd5-8dc4-cdedbca31c79)

 I copied it from version 2.0 to version 1.98 and injected its instance into Db2DatabaseSchema:
![1718178650986](https://github.com/apache/flink-cdc/assets/18555755/549439ee-921c-40c7-a1fc-711607d1b9e5)
Everything is working now!

Hope this can be a reference for you to fix this BUG.

**Reply by yuxiqian** [reply]
Cool! Could you please open a PR with this patch and testcases? Thanks!
